examples/unet_inference_3d.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import tempfile
+from glob import glob
+
+import nibabel as nib
+import numpy as np
+import torch
+import torchvision.transforms as transforms
+from ignite.engine import Engine
+from torch.utils.data import DataLoader
+
+from monai import application
+from monai.application.handlers.segmentation_saver import SegmentationSaver
+from monai.data.readers import NiftiDataset
+from monai.data.transforms import AddChannel, Rescale, ToTensor
+from monai.networks.nets.unet import UNet
+from monai.networks.utils import predict_segmentation
+from monai.utils.generateddata import create_test_image_3d
+from monai.utils.sliding_window_inference import sliding_window_inference
+
+sys.path.append("..")  # assumes the framework is found here, change as necessary
+application.config.print_config()
+tempdir = tempfile.mkdtemp()
+# tempdir = './'
+for i in range(50):
+    im, seg = create_test_image_3d(256, 256, 256)
+
+    n = nib.Nifti1Image(im, np.eye(4))
+    nib.save(n, os.path.join(tempdir, 'im%i.nii.gz' % i))
+
+    n = nib.Nifti1Image(seg, np.eye(4))
+    nib.save(n, os.path.join(tempdir, 'seg%i.nii.gz' % i))
+
+images = sorted(glob(os.path.join(tempdir, 'im*.nii.gz')))
+segs = sorted(glob(os.path.join(tempdir, 'seg*.nii.gz')))
+imtrans = transforms.Compose([Rescale(), AddChannel(), ToTensor()])
+segtrans = transforms.Compose([AddChannel(), ToTensor()])
+ds = NiftiDataset(images, segs, imtrans, segtrans, image_only=False)
+
+device = torch.device("cpu:0")
+roi_size = (64, 64, 64)
+sw_batch_size = 4
+net = UNet(
+    dimensions=3,
+    in_channels=1,
+    num_classes=1,
+    channels=(16, 32, 64, 128, 256),
+    strides=(2, 2, 2, 2),
+    num_res_units=2,
+)
+net.to(device)
+
+
+def _sliding_window_processor(_engine, batch):
+    net.eval()
+    img, seg, meta_data = batch
+    with torch.no_grad():
+        seg_probs = sliding_window_inference(img, roi_size, sw_batch_size, net, device)
+        return predict_segmentation(seg_probs)
```

Comment:
the segmentation prob to crisp mask conversion is implemented here in the workflow, after stitching the patches. so I think the network's output quantisation https://github.com/Project-MONAI/MONAI/blob/294f6e13a70fe813aa35284e25b8019a3653cc97/monai/networks/nets/unet.py#L101 should be moved into an independent post-processing module, what do you think? @ericspod @atbenmurray  (this is related to issue #44)

Reply:
I would say that people would expect a segmentation network to produce segmentations itself without processing, rather than just produce the logits. The network should be all-inclusive of its functionality, certainly if we package it with torchscript  users would expect it to have all the functionality baked in to accomplish the task.

Reply:
just noticed that this unet implementation doesn't have a final softmax/sigmoid layer for segmentation probs, which means it is a generic arch. could be used for image regression such as MR-CT synthesis? let's have a meeting early next week to discuss it #44